docs/usage/use-built-in-abstract-repository.md

```diff
@@ -7,6 +7,336 @@ nav_order: 5
+
+```csharp
+public async Task<Hero> Create(string name, string superPower, bool isAlive, bool isAvenger)
+{
+    var hero = new Hero(name, superPower, isAlive, isAvenger);
+
+    await repository.AddAsync(hero);
+
+    await respository.SaveChangesAsync();
+
+    return hero;
+}
+```
+
+Now that a Hero has been created, it's possible to retrieve that Hero using either the Hero's Id or by using a Specification. Note that since the `HeroByNameSpec` returns a single Hero entity, the Specification inherits the interface `ISingleResultSpecification` which `GetBySpecAsync` uses to constrain the return type to a single Entity.
+
+```csharp
+public class HeroByNameSpec : Specification<Hero>, ISingleResultSpecification
+{
+    public HeroByNameSpec(string name)
+    {
+        if (!string.IsNullOrEmpty(name))
+        {
+            Query.Where(h => h.Name == name);
+        }
+    }
+}
+
+public async Task<Hero> GetById(int id)
+{
+    return await repository.GetByIdAsync(id);
+}
+
+public async Task<Hero> GetByName(string name)
+{
+    var spec = new HeroByNameSpec(name);
+
+    return await repository.GetBySpecAsync(spec);
+}
+```
+
+Next, a Hero can be updated using `UpdateAsync`. `HeroService` defines a method `SetIsAlive` that takes an existing Hero and updates the IsAlive property.
+
+```csharp
+public async Task<Hero> SetIsAlive(int id, bool isAlive)
+{
+    var hero = await repository.GetByIdAsync(id);
+
+    hero.IsAlive = isAlive;
+
+    await repository.UpdateAsync(hero);
+
+    await respository.SaveChangesAsync();
+
+    return hero;
+}
+```
+
+Removing Heroes can be done either by Hero using `DeleteAsync` or by collection using `DeleteRangeAsync`.
+
+```csharp
+public async Task Delete(Hero hero)
+{
+    await repository.DeleteAsync(hero);
+
+    await respository.SaveChangesAsync();
+}
+
+public async Task DeleteRange(Hero[] heroes)
+{
+    await repository.DeleteRangeAsync(heroes);
+
+    await respository.SaveChangesAsync();
+}
+```
+
+The `RepositoryBase<T>` also provides two common Linq operations `CountAsync` and `AnyAsync`.
+
+```csharp
+public async Task SeedData(Hero[] heroes)
+{
+    // only seed if no Heroes exist
+    if (await repository.AnyAsync())
+    {
+        return;
+    }
+
+    // alternatively
+    if (await repository.CountAsync() > 0)
+    {
+        return;
+    }
+
+    foreach (var hero in heroes)
+    {
+        await repository.AddAsync(hero);
+    }
+
+    await repository.SaveChangesAsync();
+}
+```
+
+The full HeroService implementation is shown below.
+
+```csharp
+public class HeroService
+{
+    private readonly YourRepository<Hero> repository;
```

Comment:
I see it here. Normally I would make a private field `_repository`. In any case maybe show this line up above so folks know what type it is before getting here.